From d8672f90d8a489b2ec14e6743f09750b3ea1f61c Mon Sep 17 00:00:00 2001
From: Jookia <166291@gmail.com>
Date: Thu, 5 Sep 2019 23:39:22 +1000
Subject: [PATCH 15/30] ARM: dts: imx6q-novena: Pin IPUs to specific muxes

The i.MX6 Quad has two IPUs shared across LVDS, HDMI and MIPI.
Spreading the dual-channel LVDS over both IPUs won't work,
so remove the nodes that allow this sharing.
Reserve IPU1 for LVDS and IPU2 for HDMI.
---
 arch/arm/boot/dts/imx6q-novena.dts | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm/boot/dts/imx6q-novena.dts b/arch/arm/boot/dts/imx6q-novena.dts
index c59a27caf3a1..bc01d9749706 100644
--- a/arch/arm/boot/dts/imx6q-novena.dts
+++ b/arch/arm/boot/dts/imx6q-novena.dts
@@ -359,6 +359,22 @@
 	fsl,err006687-workaround-present;
 };
 
+/* reserve ipu1 for lvds, remove hdmi */
+/delete-node/&ipu1_di0_hdmi;
+/delete-node/&ipu1_di1_hdmi;
+/delete-node/&hdmi_mux_0;
+/delete-node/&hdmi_mux_1;
+
+/* reserve ipu2 for hdmi, remove lvds */
+/delete-node/&ipu2_di0_lvds0;
+/delete-node/&ipu2_di1_lvds0;
+/delete-node/&ipu2_di0_lvds1;
+/delete-node/&ipu2_di1_lvds1;
+/delete-node/&lvds0_mux_2;
+/delete-node/&lvds0_mux_3;
+/delete-node/&lvds1_mux_2;
+/delete-node/&lvds1_mux_3;
+
 &hdmi {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_hdmi_novena>;
-- 
2.20.1

